perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,265 @@
+  """
+  vms = benchmark_spec.vms
+  vm = vms[0]
+  logging.info('Preparing SciMark2 on %s', vm)
+  vm.Install('build_tools')
+  vm.Install('wget')
+  vm.InstallPackages('unzip')
+  cmds = [
+      'wget {0} -O {1}'.format(SCIMARK2_URL, SCIMARK2_ZIP),
+      '(cd {0} && rm -rf {1} && unzip {2})'.format(
+          vm_util.VM_TMP_DIR, SCIMARK2_PATH, SCIMARK2_ZIP),
+      '(cd {0} && make build_c)'.format(SCIMARK2_PATH),
+      '(cd {0} && make build_java)'.format(SCIMARK2_PATH)
+  ]
+  for cmd in cmds:
+    vm.RemoteCommand(cmd, should_log=True)
+
+
+def Run(benchmark_spec):
+  """Run SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  vms = benchmark_spec.vms
+  vm = vms[0]
+  logging.info('Running SciMark2 on %s', vm)
+  samples = []
+  if TEST_PARSE_RESULTS:
+    samples.extend(TestParseResults())
+    return samples
+  cmds = [
+      '(cd {0} && make c)'.format(SCIMARK2_PATH),
+      '(cd {0} && make java)'.format(SCIMARK2_PATH)
+  ]
+  for cmd in cmds:
+    stdout, _ = vm.RemoteCommand(cmd, should_log=True)
+    samples.extend(ParseResults(stdout))
+  return samples
+
+
+def Cleanup(benchmark_spec):
+  """Cleanup SciMark2 on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  pass
+
+
+def ParseResults(results):
+  """Result parser for SciMark2.
+
+  Sample Results (C version):
+    **                                                              **
+    ** SciMark2 Numeric Benchmark, see http://math.nist.gov/scimark **
+    ** for details. (Results can be submitted to pozo@nist.gov)     **
+    **                                                              **
+    Using       2.00 seconds min time per kenel.
+    Composite Score:         1596.04
+    FFT             Mflops:  1568.64    (N=1024)
+    SOR             Mflops:  1039.98    (100 x 100)
+    MonteCarlo:     Mflops:   497.64
+    Sparse matmult  Mflops:  1974.39    (N=1000, nz=5000)
+    LU              Mflops:  2899.56    (M=100, N=100)
+
+  Sample Results (Java version):
+
+    SciMark 2.0a
+
+    Composite Score: 1731.4467627163242
+    FFT (1024): 996.9938397943672
+    SOR (100x100):   1333.5328291027124
+    Monte Carlo : 724.5221517116782
+    Sparse matmult (N=1000, nz=5000): 1488.18620413327
+    LU (100x100): 4113.998788839592
+
+    java.vendor: Oracle Corporation
+    java.version: 1.7.0_75
+    os.arch: amd64
+    os.name: Linux
+    os.version: 3.16.0-25-generic
+
+  Args:
+    results: SciMark2 result.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  RESULT_START_C = '** SciMark2 Numeric Benchmark'
+  RESULT_START_JAVA = 'SciMark 2.0a'
+
+  SCORE_REGEX = r'\n(Composite Score):\s+(\d+\.\d+)'
+  RESULT_REGEX_C = r'\n(.+?)\s+Mflops:\s+(\d+\.\d+)(\s+\(.+?\))?'
```

Comment:
Optional: consider using extended regex syntax to support whitespace and embedded comments.

Also, using `^` together with `re.M` (== `re.MULTILINE`) is a bit more robust than matching `\n` since it matches at the start of the first line too - this may not make a difference in this case, but it's a good habit.

Example (untested):

    RESULT_REGEX_C = re.compile(r'''
      ^
      ( .+? ) \s+  # Test name
      Mflops: \s+ ( \d+\.\d+ )  # Test score
      ( \s+ \( .+? \) )?  # Optional test details
    ''', re.X | re.M)